spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -3108,6 +3147,54 @@ def run_pygments_highlighter(self):
         if isinstance(self.highlighter, sh.PygmentsSH):
             self.highlighter.make_charlist()
 
+    def _handle_goto_definition_event(self, pos):
+        """Check if goto definition can be applied and apply highlight."""
+        text = self.get_word_at(pos)
+        if (text and not sourcecode.is_keyword(to_text_string(text))):
+            if not self.__cursor_changed:
+                QApplication.setOverrideCursor(
+                                            QCursor(Qt.PointingHandCursor))
+                self.__cursor_changed = True
+            cursor = self.cursorForPosition(pos)
+            cursor.select(QTextCursor.WordUnderCursor)
+            self.clear_extra_selections('ctrl_click')
+            self.__highlight_selection(
+                'ctrl_click', cursor, update=True,
+                foreground_color=self.ctrl_click_color,
+                underline_color=self.ctrl_click_color,
+                underline_style=QTextCharFormat.SingleUnderline)
+            return True
+        else:
+            return False
+
+    def _handle_goto_uri_event(self, pos):
+        """Check if go to uri can be applied and apply highlight."""
+        uri, cursor = self.get_uri_at(pos)
+        if uri and cursor:
+            color = self.ctrl_click_color
+
+            if uri.startswith('file://'):
+                fname = uri.replace('file://', '')
+                if not osp.isfile(fname):
+                    color = QColor('red')

```

Comment:
This is a very strong color. What is you use instead `rgb(255, 80, 80)`? That's what we use in other places to signal failure.